packages/langium/src/lsp/language-server.ts

```diff
@@ -34,6 +35,9 @@ export function startLanguageServer(services: LangiumSharedServices): void {
                         supported: true
                     }
                 },
+                executeCommandProvider: commandNames && {
```

Comment:
Q: Would it only assign `{commands: commandNames}` if `commandNames ` is defined?

Reply:
A: Exactly, the `&&` is a short-circuit operator in this case. The whole operation returns `undefined` if `commandNames` is undefined, and `{commands: commandNames}` otherwise.